dpm/client/__init__.py

```diff
@@ -9,21 +9,30 @@
 class DpmException(Exception):
     pass
 
+
+class ConfigError(Exception):
+    """ The configuration passed to the client is malformed """
+    pass
+
+
 class Client(object):
 
-    def __init__(self, data_package_path='', config_path="~/.dpm/config", click=None):
+    def __init__(self, config, data_package_path='', click=None):
```

Comment:
data_package_path should come first - it is primary.

Re config: i am in two minds as to whether we want client to create config object or get passed it. From point of view of a library using this client i guess passing client in may be nice - it can also be mocked out so this way *is* nicer 😄